### PR TITLE
Improve media & text block on IE.

### LINKED ISSUE #11577: Media & Text block is not aligning in IE 11
I wonder while we know that `display: grid` is NOT supported in IE11 and will not be implemented in any possible way, why are we still pushing for it?

We can either drop the IE11 support or stick to the cutting edge CSS techniques.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/11577
Grid layout on ie has some special needs.
This PR applies a series of changes to get to a point where the main block problems on IE are mostly solved.
The autoprefixer configuration was updated to support grid prefixes.

The remaining problem on IE is that on the frontend the media & content always appear at 50% width. The way to solve this problem would require that we set an IE-specific inline style on the markup of the block. So we would be transferring more bytes over the wire for all the users to fix the problem in this browser, so I opted to leave this change out. In my option, the resizing can be seen as a progressive enhancement.

## Screenshots:

Before:

screenshot 2019-02-14 at 14 19 19

After:
screenshot 2019-02-14 at 14 18 09

### CODE REVIEW

File: packages/block-library/src/media-text/editor.scss
Comment (youknowriad) [inline packages/block-library/src/media-text/editor.scss]: Instead of adding this mse specific styles here, can we just enable `grid: true` in autoprefixer config?

Comment (jorgefilipecosta) [reply packages/block-library/src/media-text/editor.scss]: Hi @youknowriad, I agree using the autoprefixer is an awesome idea and makes the code simpler.
I did some tests, and the template areas we had were not working even with the autoprefixer, setting columns spans using grid-column: 1 / -1; also did not work but using grid-column: 1 / span 2; works.
It looks like with this changes to the CSS and using the autoprefixer things are working fine in more modern browsers and in IE11, without having IE prefixes around the code.
Thank you for the suggestion.

Comment (youknowriad) [reply packages/block-library/src/media-text/editor.scss]: I don't see the change to the autoprefixer config in the changes? Is it already enabled?

Comment (jorgefilipecosta) [reply packages/block-library/src/media-text/editor.scss]: Sorry, I missed a stash on the change, the problem was corrected.

**Comment by gziolo** [comment]
@jorgefilipecosta this branch has merge conflicts and needs to be refreshed. It seems like it never to reviewed. Can you bring it up to date and remove `Stale` label which I'm applying to make review and triage process easier?

**Comment by jorgefilipecosta** [comment]
Hi @gziolo this PR was rebased and updated to work with the new "Media & Text" functionalities like the stack on mobile.

File: packages/block-library/src/media-text/edit.js
Comment (aduth) [inline packages/block-library/src/media-text/edit.js]: Note that style attributes must be whitelisted. This was missed for `grid-template-columns` previously, and can result in invalidated blocks for non-administrator users.

See also: https://core.trac.wordpress.org/ticket/46597

We should probably add a filter to include `-ms-grid-columns` for the `safe_style_css` set in the plugin, and propose a complementing upstream Trac ticket once this is merged.

https://developer.wordpress.org/reference/hooks/safe_style_css/

**Review by aduth — DISMISSED** [review]

File: packages/block-library/src/media-text/edit.js
Comment (jorgefilipecosta) [reply packages/block-library/src/media-text/edit.js]: Hi @aduth, this PR is not changing the block save function/markup of the block. At the time (before 5.0 release) in talks with @mtias it was discussed if we should add an IE specific style to the block inline styles and we decided not to do it. The disadvantage is that on the website frontend the image always appears at 50% width, on the backend we add the correct style so it appears on the correct width.

Given that we are not adding -ms-grid-columns on the save function I don't think there is a need to update filters compared to what we are already doing on the patches available at https://core.trac.wordpress.org/ticket/46597.

Comment (aduth) [reply packages/block-library/src/media-text/edit.js]: > this PR is not changing the block save function/markup of the block

Ah, you're right. Sorry, I missed that. 

I guess it makes me wonder then what's the purported benefit, if we're only solving the issue in the editor, but not on the front-end. To me, it would seem the latter would be the more important anyways. At the very least, should it not be marked as fixing #11577 ?

Comment (jorgefilipecosta) [reply packages/block-library/src/media-text/edit.js]: Currently, the block does not work in IE (front-end and editor), the media element and the content don't event appear side by side. This PR fixes most of the problems on the front end (in the editor fixes all of them). The layout appears side by side the vertical-align options work, the background position change works, the only thing that does not work is the resizing on the front end (e.g: things appear always with the default 50% width), compared to the current state is a big improvement. Issue https://github.com/WordPress/gutenberg/issues/11577 says "Media & Text block is not aligning in IE 11" and I guess that problem is fixed the alignments work as expected, maybe we can close and add a comment specifying that just the resizing functionality does not work on the front end.

**Comment by tyrann0us** [comment]
@jorgefilipecosta, what's the status here? Thanks!
CC @aduth, @gziolo

**Comment by jorgefilipecosta** [comment]
Hi @tyrann0us, this rebased and updated. It is waiting for a review in order to be merged.

**Comment by jorgefilipecosta** [comment]
This PR was updated and should be ready just needs a review.

**Review by youknowriad — APPROVED** [review]
Seems to work well in my testing.

**Pull request merged**
